### PR TITLE
Labels.split()

### PR DESCRIPTION
**Pull request by talmo**
### Description
Add a utility method `Labels.split()` to easily generate random splits of the data.

### Types of changes

- [ ] Bugfix
- [x] New feature
- [ ] Refactor / Code style update (no logical changes)
- [ ] Build / CI changes
- [ ] Documentation Update
- [ ] Other (explain)

### Does this address any currently open issues?
N/A

### Outside contributors checklist

- [ ] Review the [guidelines for contributing](https://github.com/murthylab/sleap/wiki/Developer-Guide) to this repository
- [ ] Read and sign the [CLA](https://github.com/murthylab/sleap/blob/develop/sleap-cla.pdf) and add yourself to the [authors list](https://github.com/murthylab/sleap/blob/develop/AUTHORS)
- [ ] Make sure you are making a pull request against the **develop** branch (not *main*). Also you should start *your branch* off *develop*
- [ ] Add tests that prove your fix is effective or that your feature works
- [ ] Add necessary documentation (if appropriate)

#### Thank you for contributing to SLEAP!
:heart:

### CODE REVIEW

File: sleap/io/dataset.py [hunk 1]
```diff
@@ -740,6 +741,47 @@ def copy(self) -> "Labels":
...
+            ```
+        """
+        if len(self) == 1:
+            if copy:
+                return self.copy(), self.copy()
+            else:
+                return self, self
+
+        # Split indices.
+        if type(n) != int:
```
Comment (ariematsliah-princeton) [inline sleap/io/dataset.py:778]: check that 0.0 < n < 1.0?

File: sleap/io/dataset.py [hunk 2]
```diff
@@ -740,6 +741,47 @@ def copy(self) -> "Labels":
...
+        Example:
+            You can generate multiple splits by calling this repeatedly:
+
+            ```py
+            # Generate a 0.8/0.1/0.1 train/val/test split.
+            labels_train, labels_val_test = labels.split(n=0.8)
+            labels_val, labels_test = labels_val_test.split(n=0.5)
+            ```
+        """
+        if len(self) == 1:
```
Comment (ariematsliah-princeton) [inline sleap/io/dataset.py:771]: what if len is 0?

File: sleap/io/dataset.py [hunk 3]
```diff
@@ -740,6 +741,47 @@ def copy(self) -> "Labels":
...
+            if copy:
+                return self.copy(), self.copy()
+            else:
+                return self, self
+
+        # Split indices.
+        if type(n) != int:
+            n = round(len(self) * n)
+        n = max(min(n, len(self) - 1), 1)
+        idx_a, idx_b = train_test_split(list(range(len(self))), train_size=n)
```
Comment (ariematsliah-princeton) [inline sleap/io/dataset.py:781]: just for reproducibility sake, pass in 0 as random seed?
